src/components/Inputs.tsx

```
@@ -18,27 +18,60 @@ interface InputData {
   aud?: string;
 }
 
+interface ClaimEntry {
+  name: string;
+  value: string;
+  index: number;
+}
+
+const reducer = (state: any, action: any) => {
+  const stateCopy = [...state];
+  const handleChangeValues = (field: string, index: number, value: string) => (stateCopy[index][field] = value);
+
+  switch (action.action) {
+    case 'addEmptyClaim':
+      return [...state, { name: '', value: '' }];
+    case 'name':
+      handleChangeValues('name', action.payload.index, action.payload.value);
+      return stateCopy;
+    case 'value':
+      handleChangeValues('value', action.payload.index, action.payload.value);
+      return stateCopy;
+    case 'delete':
+      return stateCopy.filter(item => item !== stateCopy[action.index]);
+
+    default:
+      return state;
```

Is this an allowed action? If not it would be better to throw an error.